### PR TITLE
Implement ETH/68

### PR DESCRIPTION
**Pull request by jochem-brouwer**
Implements ETH/68: (note: this does NOT remove ETH/66 support)

https://github.com/ethereum/devp2p/blob/master/caps/eth.md#eth68-eip-5793-october-2022

https://eips.ethereum.org/EIPS/eip-5793

TODOs (currently super WIP obviously):

- [ ] Get a better understanding of clients net sublib
- [ ] Write ETH/68 support ALONG with ETH/66 support
- [ ] Write tests
- [ ] Run some hive tests to verify that ETH/68 works
- [ ] Side-task: check if we can directly integrate ETH/67 in it

Note: need to merge #2263 first

### CODE REVIEW

File: packages/client/src/net/protocol/ethprotocol.ts [hunk 1]
```diff
@@ -206,8 +206,38 @@ export class EthProtocol extends Protocol {
     {
       name: 'NewPooledTransactionHashes',
       code: 0x08,
-      encode: (hashes: Uint8Array[]) => hashes,
-      decode: (hashes: Uint8Array[]) => hashes,
+      // If eth protocol is eth/68, the parameter list for `NewPooledTransactionHashes` changes from
+      // `hashes: Uint8Array[]` to an tuple of arrays of `types, sizez, hashes`, where types corresponds to the
```
Comment (jochem-brouwer) [inline packages/client/src/net/protocol/ethprotocol.ts:210]: `sizez` -> `sizes`

File: packages/client/src/net/protocol/ethprotocol.ts [hunk 2]
```diff
@@ -206,8 +206,38 @@ export class EthProtocol extends Protocol {
...
+      // transaction type, sizes is the size of each encoded transaction in bytes, and the transaction hashez
```
Comment (jochem-brouwer) [inline packages/client/src/net/protocol/ethprotocol.ts:211]: `hashez` -> `hashes`

File: packages/client/src/net/protocol/ethprotocol.ts [hunk 3]
```diff
@@ -343,7 +373,7 @@ export class EthProtocol extends Protocol {
    * Protocol versions supported
    */
   get versions() {
-    return [66]
+    return [66, 67, 68]
```
Comment (jochem-brouwer) [inline packages/client/src/net/protocol/ethprotocol.ts:376]: Have to merge #2263

File: packages/client/src/rpc/modules/eth.ts [hunk 4]
```diff
@@ -974,7 +974,10 @@ export class Eth {
 
     try {
       await txPool.add(tx, true)
-      await txPool.sendNewTxHashes([tx.hash()], pool.peers)
+      await txPool.sendNewTxHashes(
```
Comment (jochem-brouwer) [inline packages/client/src/rpc/modules/eth.ts:977]: How does this work if we are on `eth/66`?

File: packages/client/src/service/txpool.ts [hunk 5]
```diff
@@ -457,11 +458,29 @@ export class TxPool {
...
       if (hashesToSend.length > 0) {
-        try {
-          await peer.eth?.request('NewPooledTransactionHashes', hashesToSend)
-        } catch (e) {
-          this.markFailedSends(peer, hashesToSend, e as Error)
-        }
+        if (
+          peer.eth !== undefined &&
+          peer.eth['versions'] !== undefined &&
+          peer.eth['versions'].includes(68)
```
Comment (jochem-brouwer) [inline packages/client/src/service/txpool.ts:464]: Ah ok the `eth/66`/`eth/68` switch is handled here.

File: packages/client/src/rpc/modules/eth.ts (hunk 4 above)
Comment (jochem-brouwer) [reply packages/client/src/rpc/modules/eth.ts:977]: Nvm handled in fullethereumservice

**Review by jochem-brouwer — COMMENTED** [review]
Some small comments.

General Q: are there still tests left to test the changed messages for `eth/66`? The tests added for `eth/68` are great!

**Comment by jochem-brouwer** [comment]
One final question regarding Eth/66 tests see my last comment. Otherwise can approve!

File: packages/client/src/service/txpool.ts (hunk 5 above)
Comment (acolytec3) [reply packages/client/src/service/txpool.ts:464]: Yup, I'll add a comment to clarify what's going on

**Review by acolytec3 — APPROVED** [review]
Approving to merge, based on @jochem-brouwer 's approval (since a lot of this is my code).

**Pull request merged**